pkg/ebpf/tracee.go

```diff
@@ -328,14 +335,57 @@ func (t *Tracee) Init() error {
 		}
 	}
 
-	// the process-tree initialize must be before we start receiving events to ensure we already have the needed data for the events that use that tree
+	// Initialize process tree before receiving events (will use it)
+
 	t.procInfo, err = procinfo.NewProcessInfo()
 	if err != nil {
 		t.Close()
 		return fmt.Errorf("error creating process tree: %v", err)
 	}
 
-	t.containers, err = containers.New(t.config.Sockets, "containers_map", t.config.Debug)
+	// Initialize cgroup filesystems
+
+	var defaultCgroup cgroup.Cgroup
+
+	isCgroupV1, err := cgroup.IsCgroupV1Default()
+	if err != nil {
+		return err
+	}
+	t.cgroupv1 = nil
+	if isCgroupV1 {
+		t.cgroupv1 = cgroup.InitCgroupV1()
+		err = t.cgroupv1.Initialize()
+		if err != nil {
+			return err // cgroupv1 is unavailable when it should be
+		}
+		defaultCgroup = t.cgroupv1 // default is cgroupv1
+	}
+
+	isCgroupV2, err := cgroup.IsCgroupV2Supported()
+	if err != nil {
+		return err
+	}
+	if !isCgroupV2 {
+		return fmt.Errorf("cgroupv2 is not supported in this environment")
+	}
+
+	t.cgroupv2 = cgroup.InitCgroupV2()
+	err = t.cgroupv2.Initialize()
+	if err != nil {
+		return err // cgroupv2 is unavailable (it needs to be for cgroup progs)
```

Comment:
I don't think we should fail here if we don't use the cgroup progs. It's one thing not to support network events on those systems, another entirely to just drop tracee as a whole.